internal/controller/template_controller_test.go

```diff
@@ -140,40 +180,57 @@ var _ = Describe("Template Controller Integration Test", func() {
 			}
 			return nil
 		}, 10*time.Second, 2*time.Second).Should(Succeed())
-
 	})
 })
 
-// Function to apply resources from an example
-func applyExampleResources(ctx context.Context, examplePath string) {
-	inputPath := filepath.Join(examplePath, "input")
-	// get all folders in the input directory
-	folders, err := utils.GetSubDirs(inputPath)
+// Returns True if the given path refers to a YAML file
+func isYAMLFile(path string) bool {
+	fileInfo, err := os.Stat(path)
 	Expect(err).NotTo(HaveOccurred())
-	for _, folder := range folders {
-		applyYAMLFilesFromDirectory(ctx, path.Join(inputPath, folder))
-	}
+
+	return !fileInfo.IsDir() && (filepath.Ext(fileInfo.Name()) == ".yaml" || filepath.Ext(fileInfo.Name()) == ".yml")
 }
 
-// Function to apply all YAML files from a directory
-func applyYAMLFilesFromDirectory(ctx context.Context, dir string) {
-	files, err := os.ReadDir(dir)
+// Get Kubernetes object defined by the YAML in the given path
+func getK8sClientObject(path string) (*unstructured.Unstructured, error) {
+	if isValidPath := isYAMLFile(path); !isValidPath {
+		return nil, fmt.Errorf("Invalid path: %s should be a YAML file", path)
+	}
+
+	content, err := os.ReadFile(path)
 	Expect(err).NotTo(HaveOccurred())
 
-	for _, file := range files {
-		if !file.IsDir() && (strings.HasSuffix(file.Name(), ".yaml") || strings.HasSuffix(file.Name(), ".yml")) {
-			filePath := filepath.Join(dir, file.Name())
+	decoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	obj := &unstructured.Unstructured{}
+	_, _, err = decoder.Decode(content, nil, obj)
+	Expect(err).NotTo(HaveOccurred())
 
-			content, err := os.ReadFile(filePath)
-			Expect(err).NotTo(HaveOccurred())
+	// Remove resourceVersion if set
+	obj.SetResourceVersion("")
 
-			decoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
-			obj := &unstructured.Unstructured{}
-			_, _, err = decoder.Decode(content, nil, obj)
-			Expect(err).NotTo(HaveOccurred())
+	return obj, nil
+}
 
-			// Remove resourceVersion if set
-			obj.SetResourceVersion("")
+// Apply all resources specified by the YAML files in the given directory, dir, as well as all sub-directories
+// Ignores paths specified in excludePaths
+func applyResourcesFromYAMLInDir(ctx context.Context, dir string, excludePaths []string) {
+	if excludePaths == nil {
```

Comment:
I didn't see this the first time, I haven't used the `slices` package much, but I think `nil` will just work as an empty slice, see: https://go.dev/play/p/o6KpQfftdPf

So hopefully you can remove this if block and it will still work.